scipy/interpolate/_rgi_cython.pyx

```diff
@@ -0,0 +1,126 @@
+# cython: language_level=3
+"""
+Cythonized routines for the RegularGridInterpolator.
+"""
+
+from libc.math cimport NAN
+import numpy as np
+cimport numpy as np
+cimport cython
+
+include "_poly_common.pxi"
+
+np.import_array()
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+@cython.initializedcheck(False)
+def evaluate_linear_2d(double_or_complex[:, :] values, # cannot declare as ::1
+                       long[:, :]indices,              # unless prior
+                       double[:, :] norm_distances,    # np.ascontiguousarray
+                       tuple grid not None,
+                       double_or_complex[:] out):
+    cdef:
+        long num_points = indices.shape[1]      # XXX: npy_intp?
+        long i0, i1, point
+        double_or_complex y0, y1, summ
+    assert out.shape[0] == num_points
+
+    if grid[1].shape[0] == 1:
+        # linear interpolation along axis=0
+        for point in range(num_points):
+            i0 = indices[0, point]
+            if i0 >= 0:
+                y0 = norm_distances[0, point]
+                summ = values[i0, 0]*(1 - y0) + values[i0+1, 0]*y0
+                out[point] = summ
+            else:
+                # xi was nan: find_interval returns -1
+                out[point] = NAN
+    elif grid[0].shape[0] == 1:
+        # linear interpolation along axis=1
+        for point in range(num_points):
+            i1 = indices[1, point]
+            if i1 >= 0:
+                y1 = norm_distances[1, point]
+                summ = values[0, i1]*(1 - y1) + values[0, i1+1]*y1
+                out[point] = summ
+            else:
+                # xi was nan: find_interval returns -1
+                out[point] = NAN
+    else:
+        for point in range(num_points):
+            i0, i1 = indices[0, point], indices[1, point]
+            if i0 >=0 and i1 >=0:
+                y0, y1 = norm_distances[0, point], norm_distances[1, point]
+
+                summ = 0.0
```

Comment:
Does the extra `m` in summ mean anything?

Reply:
Uhm. That was out of habit to avoid shadowing the built-in sum. Changed to `result` in the latest push.